src/yarp.c

```diff
@@ -2241,10 +2244,67 @@ parse_expression_prefix(yp_parser_t *parser) {
       yp_node_t *statements = parse_statements(parser, YP_CONTEXT_BEGIN);
 
       accept_any(parser, 2, YP_TOKEN_NEWLINE, YP_TOKEN_SEMICOLON);
-      expect(parser, YP_TOKEN_KEYWORD_END, "Expected `end` to close `begin` statement.");
 
-      yp_token_t end_keyword = parser->previous;
-      return yp_node_begin_node_create(parser, &begin_keyword, statements, &end_keyword);
+      yp_token_t end_keyword;
+      not_provided(&end_keyword, parser->previous.end);
+
+      yp_node_t *parent = yp_node_begin_node_create(parser, &begin_keyword, statements, NULL, &end_keyword);
+
+      yp_node_t *current = NULL;
+
+      // Parse any number of rescue clauses. This will form a linked list of if
+      // nodes pointing to each other from the top.
+      while (parser->current.type == YP_TOKEN_KEYWORD_RESCUE) {
+        parser_lex(parser);
+        yp_token_t rescue_keyword = parser->previous;
+
+        yp_token_t exception_variable;
+        not_provided(&exception_variable, parser->previous.end);
+
+        yp_node_t *statements = yp_node_statements_create(parser);
```

Comment:
I have to init statements to avoid exception in the generated code but I'll override them down the line 😞Maybe there's a better way to do that